### ISSUE TITLE
Doubling the local radiation field?

### ISSUE THREAD

**Issue by ParfenovS** [lime-rt/lime#149]
It seems like the local contribution of the current cell into intensity adds two times while integrating the radiative transfer equation along the given direction. The first time local field is added in photon procedure (see line 277 in [photon.c](https://github.com/lime-rt/lime/blob/master/src/photon.c) file). The second time this will be added in getjbar procedure (remnantSnu variable at line 390 in photon.c file). This is odd. Or maybe I just don't understand something.
If it is really an error then it can be relatively easy fixed by adding the following code after line 249 in photon.c file (at the end of if(firststep){} statement):
here=there;
continue;

**Comment by tlunttil** [comment]
Yes, I think this is a bug (I think absorption of the local cell is also doubled). I was thinking of fixing this together with other strange things in the main loop of photon() (some of them discussed in #125, but also the local vfac thing that I mentioned in some comment and maybe some others). I'm planning to write a new PR after #145 has been merged.

**Comment by ParfenovS** [comment]
@tlunttil , this is one of my latest test versions of photon and getjbar procedures [photon.c](https://dl.dropboxusercontent.com/u/83439322/photon.c). It takes into account the projection of ds onto initial direction (from PR #83). It uses physical conditions in both adjacent cells during a single step of intensity integration (as suggested in issue 125). It fixes the doubling of local radiation field.
The code is not very elegant but it works. I hope that it can help you in preparing the PR you planned.

**Comment by allegroLeiden** [comment]
Fixed in #153.
